README.md
sentry Cookbook
=================

Installs and configure Sentry realtime error logging and aggregation platform. This cookbook supports sentry 7.7.1 and above.

Requirements
------------

#### platforms
- `ubuntu` - sentry has only been tested on ubuntu

#### cookbooks
- `poise-python` - for `python`, `virtualenv` and `pip` installation and LWRP.

Attributes
----------

#### sentry::default

| Key | Type | Description | Default |
|-----|------|-------------|---------|
| `['sentry']['version']` | String | which version to install | `"8.6.0" **` |
| `['sentry']['pipname']` | String | which package to install | `"sentry[postgres]"` |
| `['sentry']['plugins']` | Array | list of plugins to install: ``` [ "sentry-irc", # No version specified ["sentry-github", "0.1.2"], # With explicit version specified ] ``` | `[["django-secure", "1.0.1"], ["django-bcrypt", "0.9.2"], ["django-sendmail-backend", "0.1.2"]]` |
| `["dependency"]["packages"]` | Array | list of packages to install | `["libxml2-dev", "libxslt1-dev", "libffi-dev",]` |
| `['sentry']['user']` | String | system user to run sentry | `"sentry"` |
| `['sentry']['group']` | String | system group to run sentry | `"sentry"` |
| `['sentry']['install_dir']` | String | full path to the sentry install directory | `"/opt/sentry/"` |
| `['sentry']['filestore_dir']` | String | full path to the sentry filestore directory | `"/opt/sentry/data"` |
| `['sentry']['config_dir']` | String | path to sentry config directory | `"/opt/sentry/etc"` |
| `['sentry']['config_file_path']` | String | path to sentry config file for Sentry < 8.0 | `"/opt/sentry/etc/config.py"` |
| `['sentry']['config_python_path']` | String | path to sentry config file for Sentry >= 8.0 | `"/opt/sentry/etc/sentry.conf.py"` |
| `['sentry']['config_yaml_path']` | String | path to sentry yaml config file for Sentry >= 8.0 | `"/opt/sentry/etc/config.yml"` |
| `['sentry']['env_d_path']` | String | path to the daemontool's env.d path for sentry configurations | `"/etc/sentry.d"` |
| `['sentry']['env_path']` | String | path to the daemontool's env path for sentry configurations | `"/etc/sentry.d/env"` |
| `['sentry']['config']['url_prefix']` | String | URL where sentry will be accessible | `"http://localhost"` |
| `['sentry']['config']['db_engine']` | String | Django class to use to connect to database | `"django.db.backends.postgresql_psycopg2"` |
| `['sentry']['config']['db_options']` | Hash | OPTIONS passed to database config | `{autocommit: true}` |
| `['sentry']['config']['web_host']` | String | IP to which sentry is listening | `"127.0.0.1"` |
| `['sentry']['config']['web_port']` | String | Port to which sentry is listening | `9000` |
| `['sentry']['config']['web_options']` | Hash | additional options used in SENTRY_WEB_OPTIONS | `{"workers": 3, secure_scheme_headers: {"X-FORWARDED-PROTO": 'https'}}` |
| `['sentry']['config']['additional_apps']` | Array | additional apps to append to INSTALLED_APPS | `["djangosecure", "django_bcrypt"]` |
| `['sentry']['config']['prepend_middleware_classes']` | Array | additional middlewares classes to prepend to MIDDLEWARE_CLASSES | `["djangosecure.middleware.SecurityMiddleware"]` |
| `['sentry']['config']['append_middleware_classes']` | Array | additional middlewares classes to append to MIDDLEWARE_CLASSES | `[]` |
| `['sentry']['config']['email_default_from']` | String | email address used in from of sent emails | `"#{node["sentry"]["user"]}@#{node[:fqdn]}"` |
| `['sentry']['config']['email_backend']` | String | EMAIL_BACKEND class to use by django | `"django.core.mail.backends.smtp.EmailBackend"` |
| `['sentry']['config']['email_host']` | String | SMTP host to use | `"localhost"` |
| `['sentry']['config']['email_port']` | String | SMTP port to use | `25` |
| `['sentry']['config']['email_use_tls']` | Boolean | Set wether to use tls for auth | `false` |
| `['sentry']['config']['email_subject_prefix']` | String | Prefix for sent emails | `nil` |
| `['sentry']['data_bag']` | String | name of the data_bag holding the sentry configuration | `"sentry"` |
| `['sentry']['data_bag_item']` | String | name of the data_bag's item holding the credentials | `"credentials"` |
| `['sentry']['use_encrypted_data_bag']` | Boolean | if the data_bag is expected to be encrypted or not | `false` |
| `['sentry']['data_bag_secret']` | String | Path to the databag secret file when using an encrypted databag, if nil the value from `Chef::Config[:encrypted_data_bag_secret]` is used. | `nil` |
| `['sentry']['config']['social_auth_redirect_is_https']` | Boolean | Set this to true if your sentry installation is accessible via HTTPs and you'rs using any of the SSO integration | `false` |
| `['sentry']['google_client_id']` | String | Google SSO client ID | `nil` |
| `['sentry']['google_client_secret']` | String | Google SSO client secret | `nil` |
| `['sentry']['github_app_id']` | String | Github integration app ID | `nil` |
| `['sentry']['github_api_secret']` | String | Github integration api secret | `nil` |
| `['sentry']['asana_client_id']` | String | Asana integration client ID | `nil` |
| `['sentry']['asana_client_secret']` | String | Asana integration client secret | `nil` |
| `['sentry']['visualstudio_app_id']` | String | Visual Studio integration app ID | `nil` |
| `['sentry']['visualstudio_app_secret']` | String | Visual Studio integration app secret | `nil` |
| `['sentry']['visualstudio_client_secret']` | String | Visual Studio integration client secret | `nil` |

****NOTE** Versions prior to 7.7.x are having trouble running sentry upgrade command. It throws below error:

	STDERR: FATAL ERROR - The following SQL query failed: ALTER TABLE "sentry_authprovider" ALTER COLUMN "config" TYPE jsonb, ALTER COLUMN 	"config" DROP NOT NULL, ALTER COLUMN "config" DROP DEFAULT; The error was: column "config" cannot be cast automatically to type jsonb HINT: 	You might need to specify "USING config::jsonb".

Usage
-----

#### Sentry credentials

Store the default admin account informations as well as the database connection
information.

Here's the expected content of such a `data_bag item`:

```json
{
  "id": "credentials",
  "admin_username": "xxxxxxx",
  "admin_password": "xxxxxxx",
  "admin_first_name": "Chef",
  "admin_last_name": "Admin",
  "admin_email": "xxxxxxx",
  "database_name": "sentry",
  "database_user": "sentry",
  "database_password": "xxxxxx",
  "database_host": "",
  "database_port": "",
  "signing_token": "xxxxxxx",
  "email_host_user": "xxxxxxx",
  "email_host_password": "xxxxxxx",
  "additional_env_vars": {}
}
```

#### To install and configure sentry

Include `sentry` or more explictly `sentry::default` in your node's `run_list`:

```json
{
  "name":"my_node",
  "run_list": [
    "recipe[sentry::default]"
  ]
}
```

Upgrade info
------------

If you come from older version < 6.4.x then you need to run the following SQL query on the database, before upgrading:

```
UPDATE sentry_project SET team_id=(SELECT id FROM sentry_team LIMIT 1) WHERE team_id IS null;
```

If you used `bcrypt` for passwords, then you run the following SQL query on the database:

```
UPDATE auth_user SET password=CONCAT('bcrypt', SUBSTR(password, 3)) WHERE password LIKE 'bc$%';
```

Contributing
------------

1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write you change
4. Write tests for your change (if applicable)
5. Run the tests, ensuring they all pass
6. Submit a Pull Request using Github

License and Authors
-------------------

Primary authors:

- Jonathan Tron (@JonathanTron)

Contributors:

- Nilanjan Roy (@nilroy)
- Matt Leick (@mattlock12)
- Luka Lüdicke (@klyonrad)

Copyright 2013, Openhood S.E.N.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# CHANGELOG for sentry

This file is used to list changes made in each version of sentry.

## 0.5.0 (2018-06-18)

* Updating Sentry to the lastest version
* Revoving old data bag syntax [#24](https://github.com/JonathanTron/chef-sentry/pull/26)[@oscar123mendoza](https://github.com/oscar123mendoza)

## 0.4.5 (2018-06-18)

* Fixed the GITHUB_API_SECRET parameter. (Nilanjan Roy) 
  [#24](https://github.com/JonathanTron/chef-sentry/pull/24)
* Added required configuration while using reverse proxy. (Nilanjan Roy)
  [#25](https://github.com/JonathanTron/chef-sentry/pull/25)

## 0.4.4 (2017-09-29)

* Allow configuration of sentry-plugins SSOs

## 0.4.3 (2017-09-28)

* Ensure SENTRY_CONF is set in runit env

## 0.4.2 (2017-09-28)

* Remove django-secure as it's imcompatible with latest django

## 0.4.1 (2017-09-28)

* Fix a bug when using encrypted data bags and `node["sentry"]["data_bag_secret"]`
  is not set.

## 0.4.0 (2017-09-28)

* Update Sentry to 8.20.0

## 0.3.2 (2017-05-26)

* Construct `default["sentry"]["config"]["celerybeat_schedule_filename"]` and
  `default["sentry"]["config"]["filestore_options"]["location"]` from
  `node["sentry"]["filestore_dir"]` (Luka Lüdicke)
  [#21](https://github.com/JonathanTron/chef-sentry/pull/21)
* Update File storage setting for Sentry 8.10 compatibility (Luka Lüdicke)
  [#20](https://github.com/JonathanTron/chef-sentry/pull/20)
  [#17](https://github.com/JonathanTron/chef-sentry/issues/17)

## 0.3.1 (2017-01-02)

* Ensure parent install directory is created too (Nilanjan Roy)
  [#14](https://github.com/JonathanTron/chef-sentry/issues/14)
* Depends on `poise-python` cookbook instead of `python` (Nilanjan Roy)
  [#15](https://github.com/JonathanTron/chef-sentry/pull/15)

## 0.3.0 (2016-08-14)

* Default to Sentry 8.6.0
* Add Sentry 8.6 support (Matt Leick)
  [#9](https://github.com/JonathanTron/chef-sentry/pull/9)

## 0.2.2 (2016-02-17)

* Add support for Sentry 8.x
* Bump default version to 7.7.1

## 0.2.1 (2015-05-09)

* Update `default["sentry"]["config"]["redis_enabled"]` to true

## 0.2.0 (2015-05-09)

* Upgrade Sentry to 7.4.3
* Add a lot more configuration to allow buffers, queues, rate limits, quotas and
  TSDB.

## 0.1.2 (2014-06-30)

* Ensure `runit_service[sentry]` is not restarted during each Chef run:
  * Don't use `sentry[postgres]` meta-package as it leaves `python_pip` LWRP
  unable to detect it's already installed.
  * Add `default['sentry']['database']['pipdeps']` with a default to install
  `psycopg2==2.4.6`.
  * Add versions to `default['sentry']['plugins']` to ensure no re-install when
  not needed: `django-secure==1.0`, `django-bcrypt==0.9.2`, `django-sendmail-backend==0.1.2`

## 0.1.1 (2014-06-06)

* Upgrade Sentry to 5.4.6

## 0.1.0  (2013-07-09)

* Initial release of sentry


test/integration/cookbooks/sentry_test/metadata.rb
name "sentry_test"

depends "sentry"
depends "redisio"
depends "postgresql", "= 4.0.6"


test/integration/cookbooks/sentry_test/recipes/default.rb
#
# Cookbook Name:: sentry_test
# Recipe:: default
#
# Copyright 2013, Openhood S.E.N.C.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

include_recipe "apt"

node.override["postgresql"]["version"] = "9.6"
node.override["postgresql"]["dir"] = "/etc/postgresql/9.6/main"
node.override["postgresql"]["client"]["packages"] = ["postgresql-client-9.6", "libpq-dev"]
node.override["postgresql"]["server"]["packages"] = ["postgresql-9.6"]
node.override["postgresql"]["contrib"]["packages"] = ["postgresql-contrib-9.6"]
node.override["postgresql"]["contrib"]["extensions"] = ["citext"]

node.override["postgresql"]["password"]["postgres"] = "test"

include_recipe "postgresql::client"
include_recipe "postgresql::contrib" # For citext extension required by sentry...
include_recipe "postgresql::ruby" # So that postgresql tests pass...
include_recipe "postgresql::server"

node.default["redisio"]["package_install"] = true
node.default["redisio"]["version"] = nil

include_recipe "redisio"
include_recipe "redisio::enable"

package "postfix"
node.override["sentry"]["config"]["email_default_from"] = "sentry@example.com"

bash "create-sentry-role" do
  user "postgres"
  code <<-EOH
echo "CREATE ROLE sentry LOGIN ENCRYPTED PASSWORD 'sentry';" | psql
  EOH
  action :run
  not_if "psql -At -F ' ' -c '\du' | egrep --quiet '^sentry '", user: "postgres"
end

bash "create-sentry-database" do
  user "postgres"
  code <<-EOH
createdb -E UTF-8 -O sentry sentry
  EOH
  action :run
  not_if "psql -At -F ' ' -l | egrep --quiet '^sentry '", user: "postgres"
end

include_recipe "sentry::default"


test/integration/cookbooks/sentry_test/recipes/s3_host.rb
#
# Cookbook Name:: wal-e_test
# Recipe:: s3_host
#
# Copyright 2013, Openhood S.E.N.C.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

# We assume the postgresql group was already created
group "postgres"

include_recipe "wal-e::default"


recipes/_install.rb
#
# Cookbook Name:: sentry
# Recipe:: _install
#
# Copyright 2013, Openhood S.E.N.C.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

if node["sentry"]["manage_redis"]
  include_recipe "sentry::_redis"
end

include_recipe "poise-python"

sentry_user = node["sentry"]["user"]
sentry_group = node["sentry"]["group"]

group sentry_group

user sentry_user do
  group sentry_group
  home node["sentry"]["install_dir"]
  action [:create, :manage]
end

directory node["sentry"]["install_dir"] do
  owner sentry_user
  group sentry_group
  recursive true
end

directory node["sentry"]["filestore_dir"] do
  owner sentry_user
  group sentry_group
  recursive true
end

# Create a virtualenv for sentry
python_virtualenv node["sentry"]["install_dir"] do
  user sentry_user
  group sentry_group
  action :create
end

# Install deps packages to build sentry's deps
node["sentry"]["dependency"]["packages"].each do |name|
  package name
end

# Install sentry via pip in virtualenv
python_package node["sentry"]["pipname"] do
  virtualenv node["sentry"]["install_dir"]
  version node["sentry"]["version"]
  user sentry_user
  group sentry_group
end

# Install database pip dependency
node["sentry"]["database"]["pipdeps"].each do |dep|
  dep_name, dep_version = dep

  python_package dep_name do
    virtualenv node["sentry"]["install_dir"]
    version dep_version
    user sentry_user
    group sentry_group
  end
end

# Install additional plugins via pip in virtualenv
node["sentry"]["plugins"].each do |plugin|
  plugin_name, plugin_version = plugin

  python_package plugin_name do
    virtualenv node["sentry"]["install_dir"]
    version plugin_version
    user sentry_user
    group sentry_group
  end
end


recipes/_service.rb
#
# Cookbook Name:: sentry
# Recipe:: _service
#
# Copyright 2013, Openhood S.E.N.C.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

include_recipe "runit"

subscribes_resources = [
   "template[#{node["sentry"]["config_file_path"]}]",
   "python_package[#{node["sentry"]["pipname"]}]"
 ] +
   node["sentry"]["database"]["pipdeps"].map do |dep|
     dep_name, _ = dep
     "python_package[#{dep_name}]"
   end +
   node["sentry"]["plugins"].map do |plugin|
     plugin_name, _ = plugin
     "python_package[#{plugin_name}]"
   end

# sentry 'start' and 'celery' were deprecated in 8.0 in favor of
# sentry 'run < web/worker/cron >'
if node['sentry']['version'] .split('.')[0].to_i >= 8

  runit_service "sentry-web" do
    options({
      virtualenv_activate: "#{node["sentry"]["install_dir"]}/bin/activate",
      env_path: "#{node['sentry']['env_path']}",
      user: node["sentry"]["user"],
      group: node["sentry"]["group"],
      sentry_cmd: "#{node["sentry"]["install_dir"]}/bin/sentry"
    })
    sv_timeout 180
    retries 2
    retry_delay 5
    action [:enable, :start]
    subscribes_resources.each do |res|
      subscribes :restart, res, :delayed
    end
  end

  runit_service "sentry-worker" do
    options({
      virtualenv_activate: "#{node["sentry"]["install_dir"]}/bin/activate",
      env_path: "#{node['sentry']['env_path']}",
      user: node["sentry"]["user"],
      group: node["sentry"]["group"],
      sentry_cmd: "#{node["sentry"]["install_dir"]}/bin/sentry"
    })
    sv_timeout 10
    retries 2
    retry_delay 5
    action [:enable, :start]
    subscribes_resources.each do |res|
      subscribes :restart, res, :delayed
    end
  end

  runit_service "sentry-cron" do
    options({
      virtualenv_activate: "#{node["sentry"]["install_dir"]}/bin/activate",
      env_path: "#{node['sentry']['env_path']}",
      user: node["sentry"]["user"],
      group: node["sentry"]["group"],
      sentry_cmd: "#{node["sentry"]["install_dir"]}/bin/sentry"
    })
    sv_timeout 10
    retries 2
    retry_delay 5
    action [:enable, :start]
    subscribes_resources.each do |res|
      subscribes :restart, res, :delayed
    end
  end
else # using sentry commands: 'start' & 'celery'
  runit_service "sentry" do
    options({
      virtualenv_activate: "#{node["sentry"]["install_dir"]}/bin/activate",
      user: node["sentry"]["user"],
      group: node["sentry"]["group"],
      env_path: node["sentry"]["env_path"],
      sentry_cmd: "#{node["sentry"]["install_dir"]}/bin/sentry",
      config_path: node["sentry"]["config_file_path"],
    })
    retries 2
    retry_delay 5
    action [:enable, :start]
    subscribes_resources.each do |res|
      subscribes :restart, res, :delayed
    end
  end

  runit_service "sentry_queue" do
    options({
      virtualenv_activate: "#{node["sentry"]["install_dir"]}/bin/activate",
      user: node["sentry"]["user"],
      group: node["sentry"]["group"],
      env_path: node["sentry"]["env_path"],
      sentry_cmd: "#{node["sentry"]["install_dir"]}/bin/sentry",
      config_path: node["sentry"]["config_file_path"],
    })
    retries 2
    retry_delay 5
    action [:enable, :start]
    subscribes_resources.each do |res|
      subscribes :restart, res, :delayed
    end
  end

end


attributes/default.rb
#
# Cookbook Name:: sentry
# Attribute:: default
#
# Copyright 2013, Openhood S.E.N.C.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

include_attribute "poise-python::default"
default["sentry"]["version"] = "9.0.0"
default["sentry"]["user"] = "sentry"
default["sentry"]["group"] = "sentry"
default["sentry"]["pipname"] = "sentry"
default["sentry"]["database"]["pipdeps"] = []
default["sentry"]["plugins"] = [
  ["sentry-plugins", "9.0.0"],
  ["django-bcrypt", "0.9.2"],
  ["django-sendmail-backend", "0.1.2"],
]
# dependencies per: https://docs.getsentry.com/on-premise/server/installation/python/
default["sentry"]["dependency"]["packages"] = [
  "python-dev",
  "gcc",
  "libjpeg-dev",
  "libxml2-dev",
  "libxslt-dev",
  "libxslt1-dev",
  "libffi-dev",
  "libyaml-dev",
  "libpq-dev", # not in docs listed but blocks sentry install
]
default["sentry"]["install_dir"] = "/opt/sentry"
default["sentry"]["filestore_dir"] = "/opt/sentry/data"
default["sentry"]["config_dir"] = "#{node["sentry"]["install_dir"]}/etc"
default["sentry"]["config_file_path"] = "#{node["sentry"]["config_dir"]}/config.py"
default["sentry"]["config_python_path"] = "#{node["sentry"]["config_dir"]}/sentry.conf.py"
default["sentry"]["config_yaml_path"] = "#{node["sentry"]["config_dir"]}/config.yml"

# ENV variables for runit
default["sentry"]["env_d_path"] = "/etc/sentry.d"
default["sentry"]["env_path"] = "#{node["sentry"]["env_d_path"]}/env"

default["sentry"]["config"]["db_engine"] = "django.db.backends.postgresql_psycopg2"
default["sentry"]["config"]["db_options"] = {autocommit: true}
default["sentry"]["config"]["admin_email"] = ""

# web server
default["sentry"]["config"]["allow_registration"] = false
default["sentry"]["config"]["allow_origin"] = false
default["sentry"]["config"]["beacon"] = false
default["sentry"]["config"]["public"] = false
default["sentry"]["config"]["web_host"] = "127.0.0.1"
default["sentry"]["config"]["web_port"] = 9000
default["sentry"]["config"]["secure_proxy_ssl_header"] = false
default["sentry"]["config"]["web_options"] = {
  "workers" => 3,
  "secure_scheme_headers" => {
    "X-FORWARDED-PROTO" => "https"
  }
}
default["sentry"]["config"]["url_prefix"] = "http://localhost:#{node["sentry"]["config"]["web_port"]}"
default["sentry"]["config"]["force_script_name"] = false

# smtp
default["sentry"]["config"]["smtp_host"] = '0.0.0.0'
default["sentry"]["config"]["smtp_port"] = '5587'
default["sentry"]["config"]["smtp_hostname"] = 'reply.getsentry.com'

default["sentry"]["config"]["email_default_from"] = "#{node["sentry"]["user"]}@#{node[:fqdn]}"
default["sentry"]["config"]["email_backend"] = "django.core.mail.backends.smtp.EmailBackend"
default["sentry"]["config"]["email_host"] = "localhost"
default["sentry"]["config"]["email_port"] = "25"
default["sentry"]["config"]["email_use_tls"] = false
default["sentry"]["config"]["email_subject_prefix"] = nil
default["sentry"]["config"]["email_list_namespace"] = "localhost"
default["sentry"]["config"]["email_enable_replies"] = false
default["sentry"]["config"]["additional_apps"] = ["django_bcrypt"]
default["sentry"]["config"]["prepend_middleware_classes"] = []
default["sentry"]["config"]["append_middleware_classes"] = []
# general
default["sentry"]["config"]["use_big_ints"] = true
default["sentry"]["config"]["single_organization"] = true
default["sentry"]["config"]["debug"] = false
# Redis config
default["sentry"]["config"]["redis_enabled"] = true
default["sentry"]["config"]["redis_config"]["hosts"][0]["name"] = "default"
default["sentry"]["config"]["redis_config"]["hosts"][0]["host"] = "127.0.0.1"
default["sentry"]["config"]["redis_config"]["hosts"][0]["port"] = "6379"
# Cache config
default["sentry"]["config"]["cache"] = "sentry.cache.redis.RedisCache"
# Queue config
default["sentry"]["config"]["celery_always_eager"] = false # true will disable queue usage
default["sentry"]["config"]["broker_url"] = "redis://localhost:6379"
default["sentry"]["config"]["celeryd_concurrency"] = 1
default["sentry"]["config"]["celery_send_events"] = false
default["sentry"]["config"]["celerybeat_schedule_filename"] = "#{node["sentry"]["filestore_dir"]}/celery_beat_schedule"
#digests config
default["sentry"]["config"]["digests"] = 'sentry.digests.backends.redis.RedisBackend'
default["sentry"]["config"]["ratelimiter"] = "sentry.ratelimits.redis.RedisRateLimiter"
default["sentry"]["config"]["buffer"] = "sentry.buffer.redis.RedisBuffer"
default["sentry"]["config"]["quotas"] = "sentry.quotas.redis.RedisQuota"
default["sentry"]["config"]["tsdb"] = "sentry.tsdb.redis.RedisTSDB"
# Filestore config
default["sentry"]["config"]["filestore"] = "django.core.files.storage.FileSystemStorage"
default["sentry"]["config"]["filestore_options"]["location"] = node["sentry"]["filestore_dir"]
# data sampling
default["sentry"]["config"]["sample_data"] = false

default["sentry"]["data_bag"] = "sentry"
default["sentry"]["data_bag_item"] = "credentials"
default["sentry"]["use_encrypted_data_bag"] = false

default["sentry"]["manage_redis"] = true

# sentry-plugins integrations
default["sentry"]["config"]["social_auth_redirect_is_https"] = false

# google sso
default["sentry"]["google_client_id"] = nil
default["sentry"]["google_client_secret"] = nil

# github sso
default["sentry"]["github_app_id"] = nil
default["sentry"]["github_api_secret"] = nil

# asana sso
default["sentry"]["asana_client_id"] = nil
default["sentry"]["asana_client_secret"] = nil

# visual studio sso
default["sentry"]["visualstudio_app_id"] = nil
default["sentry"]["visualstudio_app_secret"] = nil
default["sentry"]["visualstudio_client_secret"] = nil
